### ISSUE TITLE
Using zcl_excel_writer_2007 more than once causes an abend

### ISSUE THREAD

**Issue by konijn** [abap2xlsx/abap2xlsx#457]
Greetings,

this code will abend when writing out the 2nd excel file:
```abap
FORM update_cache USING
                    p_excel1 TYPE REF TO zcl_excel
                    p_excel2 TYPE REF TO zcl_excel
                  CHANGING
                    p_cache TYPE tytt_excel_cache.

  DATA: wa_cache TYPE ty_excel_cache.
  DATA: cl_writer   TYPE REF TO zif_excel_writer.

  wa_cache-cc = p_cc.
  CREATE OBJECT cl_writer TYPE zcl_excel_writer_2007.
  wa_cache-file1 = cl_writer->write_file( p_excel1 ).
  wa_cache-file2 = cl_writer->write_file( p_excel2 ). "<-- This will abend

  APPEND wa_cache TO p_cache.

ENDFORM.
```
Abend info:
```text
Short Text
    Error while inserting or changing rows in a sorted table

LOOP AT lt_cell_data ASSIGNING <fs_sheet_content> where data_type = 's'.
-    lv_sytabix = sy-tabix - 1.
  lv_sytabix = lv_count.
  MOVE lv_sytabix                    TO ls_shared_string-string_no.
  MOVE <fs_sheet_content>-cell_value TO ls_shared_string-string_value.
  MOVE <fs_sheet_content>-data_type TO ls_shared_string-string_type.
  APPEND ls_shared_string TO shared_strings.
  add 1 to lv_count.
ENDLOOP.
```
I guess that in my case sheet1 and sheet2 share some common strings, and this code does an append without checking for pre-existing data. You should either clear first ( is what I would do) , or do a check first. I would clear first since you don't want o write out shared strings from prior workbooks.

**Comment by sandraros** [comment]
Thank you for raising this issue.

A possible fix would be to reinitialize all the attributes of the writer each time it is used. Help welcome.

The workaround is to reinstantiate the writer each time an Excel file is to be rendered:
```abap
CREATE OBJECT cl_writer TYPE zcl_excel_writer_2007.
wa_cache-file1 = cl_writer->write_file( p_excel1 ).
CREATE OBJECT cl_writer TYPE zcl_excel_writer_2007. "<-- workaround
wa_cache-file2 = cl_writer->write_file( p_excel2 ). "<-- This will NOT abend
```

**Comment by konijn** [comment]
Wow, 5 years ;) 
If you need help with the clearing (understandable), you could just add a flag 'ran_already', and if the writer gets used a second time, throw a catchable error. That is way sexier than an abend ;)

**Comment by sandraros** [comment]
I couldn't reproduce the issue, so I looked a little bit further and I saw that it was solved 2 years ago by #654, so I think it's more simple to close. Please reopen if you don't agree. Thank you.
